### ISSUE TITLE
code editor issues when textension is enabled

### ISSUE THREAD

**Issue by Hasib345** [K-410/textension#5]
I cannot scroll from the editor window or use tabs when textension is enabled
can you check why textension is causing issue with code editor

**Comment by K-410** [comment]
Thanks for the report.

Textension is currently being rewritten so it doesn't make much sense to update the old version to make it compatible with code editor.

There's no ETA yet, so I'll keep this issue open and address it once the new version of Textension has been pushed.

**Comment by Hasib345** [comment]
Thank for relpying I checked the code and found problem is cause due to cursor set option but was not able to fix it .

**Comment by Oxicid** [comment]
[K-410](https://github.com/K-410) Hi, I don't know where best to post, I just don't know your email address. Looking at your code, I see that you are good with ctypes, I would like to ask why my code causes an error (null pointer access)? I don't understand if 

it's a context sensitive structure or I'm missing something. 

I just wanted to simplify the work of an addon developer, but I can't get access to the variable. 
[VoronoiLinker get socket loc.txt](https://github.com/K-410/textension/files/12566888/VoronoiLinker.get.socket.loc.txt)

**Comment by K-410** [comment]
> [K-410](https://github.com/K-410) Hi, I don't know where best to post, I just don't know your email address. Looking at your code, I see that you are good with ctypes, I would like to ask why my code causes an error (null pointer access)? I don't understand if
> 
> it's a context sensitive structure or I'm missing something.

You were missing a some padding and the `ns` member of `bNodeSocket` must be a struct. C++ subclasses have inherent padding from bases.
Blender's c++ vector is mostly guesswork and depends on debug/release and compiler used. If you just need the socket location, you can omit everything else after it.
https://gist.github.com/K-410/14505d5c0c6c565c2ed445958b8ff4ff

**Comment by Oxicid** [comment]
[K-410](https://github.com/K-410) Ea, everything works fine, although the developers change these structures all the time, but I had everything working on version 3.5 and 4.0. 

Yes about Vector you are right, I don't need them, I didn't originally write them, I just discovered the world of possibilities through ctypes this week. Now I will know that for substructures it is necessary to implement fields and that something is copied during inheritance. 

About NonCopyable, NonMovable, it's just that I still don't fully understand C++ syntax, and I didn't see (still don't see) fields there, as if there is nothing but methods.

Thank you very much, but for some reason I didn't notice you before, and the textension query doesn't lead to anything. I tried to install your addon, at first it turned out that btypes did not download, and then it crashes, well I realized that the addon is under general repair, I will definitely use it, as I need something like this.
